content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,580 @@
+        cart.addProductToCartByCustomer(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+
+        assertEquals(new BigDecimal( 3100), cart.calculatePrice());
+        customerDao=null;
+    }
+    @Test
+    void addProductToCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart=null;
+    }
+    @Test
+    void removeProductsFromCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        System.out.println(cart);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart.removeProductsFromCartByAdmin(1L,product);
+        System.out.println(cart);
+        assertEquals(2,cart.totalNumberOfItems());
+
+    }
+    @Test
+    void checkAllTheProductsInTheCartByCustomer(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Product product3 = new Product("3", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product3);
+        System.out.println(  cart.checkAllTheProductsInTheCartByCustomer(1L));
+        cart=null;
+    }
+    @Test
+    void checkAllRegisteredAdmin(){
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        System.out.println(dataService.getAllAdmin());
+    }
+}
+```
+### Using Postman to verify the output on the web
+![shopping_api](api.JPG)
+![shopping_api1](api2.JPG)
+![shopping_api2](payment.JPG)
+#### Conclusion
+In the end, we've seen how to code to an interface, how to write a unit test, how to create an endpoint, and how to use postman to view our endpoint. Also, we discussed how we used field dependency injection and not constructor dependency injection which will still suffice. Furthermore, we also added products to our cart and the cart could calculate the cost of products in the cart since the products have an attribute of price. 
```

Comment:
You never talked about field dependency injection vs constructor dependency injection.